templates/al2023/provisioners/install-worker.sh

```diff
@@ -63,7 +63,8 @@ sudo dnf install -y \
   unzip \
   wget \
   mdadm \
-  pigz
+  pigz \
+  python3-dnf-plugin-versionlock
```

Comment:
Installing this for every variant, because we also need to lock `amazon-ec2-net-utils` in a later PR.